MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -7699,5 +7703,28 @@ private void stp4cs()
             Module.Memory.SetArray(stringPointer, result.ToArray());
             Registers.SetPointer(stringPointer);
         }
+        
+        /// <summary>
+        ///     filelength - gets file size in bytes
+        ///
+        ///     Signature: long filelength(int handle);
+        /// </summary>
+        private void filelength()
+        {
+            var fileHandle = GetParameter(0);
+            
+            if (FilePointerDictionary.TryGetValue(fileHandle,out var filePointer))
+            {
```

Comment:
Before the `if`, you can declare `var result = -1L` and if the value exists, set `result = filePointer.Length` (no need to cast as `result` will already be a `long`).

This way you can just have a single statement setting the registers to the value of `result`. Reduces complexity of the method 👍

Reply:
Nice ok, that made is a lot simpler 👍